mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp

```diff
@@ -1833,102 +1889,122 @@ static LogicalResult populateValidationLogic(
                              inputHostAllocOp, inputDimension, convertFuncs);
-      cpuOutputHostAllocOp =
+      verifierOutputHostAllocOp =
           allocAndCopyTensor(module, builder, block, mcpuMemCopy5DFuncOp,
                              outputHostAllocOp, outputDimension, convertFuncs);
     }
   }
 
   // Populate host validation logic
-  auto cpuConvFuncOp = createCPUConvolution(module, builder, filterMemRefType,
-                                            inputMemRefType, outputMemRefType);
-
-  // Emit conv2d_host function call.
-  auto cpuConvCallOp = builder.create<CallOp>(
-      builder.getUnknownLoc(), cpuConvFuncOp,
-      ValueRange{cpuFilterHostAllocOp, cpuInputHostAllocOp,
-                 cpuOutputHostAllocOp});
-  block->push_back(cpuConvCallOp);
+  if (populateValidationWithGPU.getValue() &&
+      (xdlopsV2.getValue() || dataType != builder.getF32Type())) {
+    // Verify with GPU convolution of f32
+    auto verifierConvFuncOp = launchGPUConvolution(
+        module, builder, floatType, verifierFilterHostAllocOp,
+        verifierInputHostAllocOp, verifierOutputHostAllocOp,
+        populateValidateEntryPoint);
+
+    // Emit gpu_conv_verifier function call
+    auto verifierConvCallOp = builder.create<CallOp>(
+        builder.getUnknownLoc(), verifierConvFuncOp,
+        ValueRange{verifierFilterHostAllocOp, verifierInputHostAllocOp,
+                   verifierOutputHostAllocOp});
+    block->push_back(verifierConvCallOp);
+  } else {
+    // Verity with CPU convolution of f32
+    auto cpuConvFuncOp = createCPUConvolution(
+        module, builder, filterMemRefType, inputMemRefType, outputMemRefType);
+
+    // Emit conv2d_host function call.
+    auto cpuConvCallOp = builder.create<CallOp>(
+        builder.getUnknownLoc(), cpuConvFuncOp,
+        ValueRange{verifierFilterHostAllocOp, verifierInputHostAllocOp,
+                   verifierOutputHostAllocOp});
+    block->push_back(cpuConvCallOp);
+  }
 
-  mlir::Value cpuResults;
+  mlir::Value verifierResults;
   if (operation.getValue() == "conv2d" ||
       operation.getValue() == "conv2d_dummy") {
-    cpuResults = cpuOutputHostAllocOp;
+    verifierResults = verifierOutputHostAllocOp;
   } else if (operation.getValue() == "conv2d_bwd_data") {
-    cpuResults = cpuInputHostAllocOp;
+    verifierResults = verifierInputHostAllocOp;
   } else if (operation.getValue() == "conv2d_bwd_weight") {
-    cpuResults = cpuFilterHostAllocOp;
+    verifierResults = verifierFilterHostAllocOp;
   }
 
   // Convert CPU results
-  mlir::Value cpuConvertedResults;
+  mlir::Value verifierConvertedResults;
   if (dataType == builder.getF32Type())
-    cpuConvertedResults = cpuResults;
+    verifierConvertedResults = verifierResults;
   else
-    cpuConvertedResults =
-        getConvertedCpuResults(module, builder, block, cpuResults,
+    verifierConvertedResults =
+        getConvertedCpuResults(module, builder, block, verifierResults,
```

Comment:
Why no braces around these blocks?

Reply:
Because I'm not good at being consistent.

Reply:
Happens - I just don't want us to be an a `goto fail` situation.